src/contracts/examples/fastlane-online/SolverGateway.sol

```diff
@@ -201,6 +205,10 @@ contract SolverGateway is OuterHelpers {
         uint256 _score =
             _getWeightedScore(swapIntent, solverOp, totalGas, msg.value, maxFeePerGas, _solverOps.length, aData);
 
+        // Check can be grokked more easily in the following format:
+        //      solverOpScore    _cumulativeScore (unweighted)
+        // if  -------------- >  ------------------------------  * 2
+        //      solverOpGas             totalGas
         if (_score * totalGas > _cumulativeScore * solverOp.gas * 2) {
```

Comment:
Why is cumulative score * solverOp.gas multiplied by 2? It makes sense to me now other than that

Reply:
Cumulative score hasn't been weighted by how much gas the solvers are using - it's weighted by the total gas available to all solvers.

The *2 means that when the gas is halfway used by solvers, a new solver must have a score that is equal to the average of the current ones (without the *2 a new solver would be accepted if they had a score that's half as good as the average of the current ones)

The idea is to make it harder and harder for txs that are more likely to be spam to get included as the gas gets filled up by solvers.